src/main/java/de/tum/in/www1/artemis/service/notifications/TutorialGroupNotificationService.java

```diff
@@ -0,0 +1,99 @@
+import static de.tum.in.www1.artemis.service.notifications.NotificationSettingsCommunicationChannel.EMAIL;
+
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.springframework.messaging.simp.SimpMessageSendingOperations;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.enumeration.tutorialgroups.TutorialGroupRegistrationType;
+import de.tum.in.www1.artemis.domain.notification.TutorialGroupNotification;
+import de.tum.in.www1.artemis.domain.tutorialgroups.TutorialGroup;
+import de.tum.in.www1.artemis.domain.tutorialgroups.TutorialGroupRegistration;
+import de.tum.in.www1.artemis.repository.tutorialgroups.TutorialGroupNotificationRepository;
+import de.tum.in.www1.artemis.repository.tutorialgroups.TutorialGroupRegistrationRepository;
+import de.tum.in.www1.artemis.service.MailService;
+
+@Service
+public class TutorialGroupNotificationService {
+
+    private final TutorialGroupNotificationRepository tutorialGroupNotificationRepository;
+
+    private final TutorialGroupRegistrationRepository tutorialGroupRegistrationRepository;
+
+    private final SimpMessageSendingOperations messagingTemplate;
+
+    private final NotificationSettingsService notificationSettingsService;
+
+    private final MailService mailService;
+
+    public TutorialGroupNotificationService(TutorialGroupNotificationRepository tutorialGroupNotificationRepository,
+            TutorialGroupRegistrationRepository tutorialGroupRegistrationRepository, SimpMessageSendingOperations messagingTemplate,
+            NotificationSettingsService notificationSettingsService, MailService mailService) {
+        this.tutorialGroupNotificationRepository = tutorialGroupNotificationRepository;
+        this.tutorialGroupRegistrationRepository = tutorialGroupRegistrationRepository;
+        this.messagingTemplate = messagingTemplate;
+        this.notificationSettingsService = notificationSettingsService;
+        this.mailService = mailService;
+    }
+
+    /**
+     * Notify registered students about a deleted tutorial group
+     *
+     * @param tutorialGroup the deleted tutorial group
+     */
+    public void notifyAboutTutorialGroupDeletion(TutorialGroup tutorialGroup) {
+        saveAndSend(createTutorialGroupNotification(tutorialGroup, TUTORIAL_GROUP_DELETED), true);
+    }
+
+    /**
+     * Notify registered students about an updated tutorial group
+     *
+     * @param tutorialGroup the updated tutorial group
+     * @param notifyTutor   whether the tutor should be notified about the update via email
+     */
+    public void notifyAboutTutorialGroupUpdate(TutorialGroup tutorialGroup, boolean notifyTutor) {
+        saveAndSend(createTutorialGroupNotification(tutorialGroup, TUTORIAL_GROUP_UPDATED), notifyTutor);
```

Comment:
Must: Do not notify the user about changes to the tutorial group they initiated themselves.

Optional: I do not think this notification makes sense from a user perspective right now. It should either include what got updated (only important changes such as time or place), or be removed.

Reply:
I solved it now as we have it with exercises. The user can enter an optional notification text in the update form, if he/she wants a notification to be sent: 
![image](https://user-images.githubusercontent.com/29718932/196643485-705921f9-95ba-4003-af6f-6d9510f8fd19.png)

Reply:
Did it in 1ec6e2b257b4d09878415389993838bba7e417ea

Reply:
![image](https://user-images.githubusercontent.com/29718932/196691377-60f2dafe-89f5-434d-9867-da1b24d450b4.png)